### PR TITLE
Hotfix standardize logging 20

### PR DESCRIPTION
**Pull request by nikola-rados**
Logging has been changed to create a more usable output.  The logger now uses a [JSON formatter](https://github.com/madzak/python-json-logger) to output a dictionary containing a message with optional extra information.  Furthermore, log levels were changed where necessary with a comment being added to  the logging config detailing standard use.  Finally a timer function was added to track the data processing/insertion time for each module.  The result is given as insertions/sec and is listed with other metrics at the end of processing.

### CODE REVIEW

**Comment by nikola-rados** [comment]
I did a rebase to fix some of the issues that were occuring in travis.  This request is ready for review!

File: crmprtd/db.py [hunk 1]
```diff
@@ -39,34 +39,40 @@ def mass_insert_obs(sesh, obs, log=None):
...
             # Create a nested SAVEPOINT context manager to rollback to in the
             # event of unique constraint errors
-            log.debug("New SAVEPOINT for 1 obs")
+            log.debug("New savepoimt for singe observation")
             with sesh.begin_nested():
                 sesh.add(obs[0])
         except IntegrityError as e:
-            log.debug("Failure: already exists: %s %s", obs, e)
+            log.warning("Failure, observation already exists",
+                        extra={'obs': obs, 'exception': e})
```
[the remark this answers is not in the document]
Comment (nikola-rados) [reply crmprtd/db.py]: I'm not actually sure what gets logged, I'll have to investigate.

File: crmprtd/ec.py [hunk 2]
```diff
@@ -65,37 +73,41 @@ def __init__(self, et, sesh, threshold):
...
-                    tostring(member, pretty_print=True)))
-
-        log.info("Finished processing {mp}/{mt} members and inserted "
-                 "{oi}/{ot} insertable observations with {oe} already existing"
-                 .format(mp=self._members_processed,
-                         mt=self._members,
-                         oi=self._obs_insertions,
-                         ot=self._obs,
-                         oe=self._obs_existing))
+                log.exception("Error processing member, saved in logs")
```
[the remark this answers is not in the document]
Comment (nikola-rados) [reply crmprtd/ec.py]: As far as I understand it, using log.exception will output the stack trace automatically.

File: crmprtd/data/logging.yaml [hunk 3]
```diff
@@ -1,13 +1,24 @@
+# Log level standard usage
```
[the remark this answers is not in the document]
Comment (nikola-rados) [reply crmprtd/data/logging.yaml]: Got it, I will change the levels where you have mentioned.

File: crmprtd/ec.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jameshiebert) [reply crmprtd/ec.py]: Perfect!

File: crmprtd/db.py [hunk 4]
```diff
@@ -39,34 +39,40 @@ def mass_insert_obs(sesh, obs, log=None):
...
             a, b = mass_insert_obs(sesh, a, log), mass_insert_obs(sesh, b, log)
-            log.debug("Returning combined %d + %d = %d", a, b, a + b)
+            combined = a + b
+            log.debug("Returning from split call", extra={'a_split': a,
+                                                          'b_split': b,
+                                                          'both': combined})
             return a + b
         else:
-            log.debug("Success for %s obs", len(obs))
+            log.debug("Successfully inserted observations",
```
[the remark this answers is not in the document]
Comment (nikola-rados) [reply crmprtd/db.py]: As far as I could tell the ```crmrtd/db.py``` method ```mass_insert_obs``` is only used by ```wamr.py``` and so I logged the number of insertions after the method executes.

```log.info("Starting a mass insertion", extra={'num_obs': len(obs)})
        start_time = timer()
        n_insertions = mass_insert_obs(sesh, obs, log)
        run_time = timer(start_time)
        log.info("Data processed and inserted",
                 extra={'num_insertions': n_insertions,
                        'insertions_sec': (n_insertions/run_time)})```

File: scripts/moti_hourly.py [hunk 5]
```diff
@@ -33,6 +34,11 @@ def main(args):
     log = logging.getLogger('crmprtd.moti')
     if args.log_level:
         log.setLevel(args.log_level)
+    # json logger
```
[the remark this answers is not in the document]
Comment (nikola-rados) [reply scripts/moti_hourly.py]: Yes, this code existed in a bunch of modules while I was testing the json-logger.  I thought I had gotten rid of all of it after I changed the config file.

File: crmprtd/db.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nikola-rados) [reply crmprtd/db.py]: So after testing this I found that the output of the 'exception' extra is the exception message.

**Comment by nikola-rados** [comment]
@jameshiebert the requested changes have been made and pushed.  Let me know if there are any further changes needed.

**Comment by jameshiebert** [comment]
Looks great to me! I'll go ahead and merge this into master. Can you take on merging this branch into `dev`?

**Comment by nikola-rados** [comment]
Yes, I can merge it.

**Pull request merged**